packaging/build_rpm.py

```diff
@@ -336,22 +338,46 @@ def install_mock():
 
 
 def check_mock_config():
-    # allow network access during the build
-    # (we have to download packages from pypi)
-    mock_config = check_output(
-        MOCK + ['--debug-config',
-        ]).splitlines()
-
-    for line in mock_config:
-        if all(s in line for s in ('rpmbuild_networking', 'True')):
-            break
-    else:
-        logger.info('configuring rpmbuild_networking')
-        check_call(
-                '''echo "config_opts['rpmbuild_networking'] = True" '''
-                '| sudo -n tee -a /etc/mock/site-defaults.cfg',
-                shell=True,
-                )
+    """
+    allow network access during the build and set the %dist macro
+    (we have to download packages from pypi)
+    """
+    mock_config = check_output(MOCK + ['--debug-config', '-qq'])
+
+    config_opts = {}
+    config = {'config_opts': config_opts}
+    exec mock_config in config, config
+
+    options = [
+        "config_opts['rpmbuild_networking'] = True",
```

Comment:
Why are we parsing those as strings, instead of having a dict? E.g.
```
options = {
  "config_opts['rpmbuild_networking']": "True",
  "config_opts['macros']['%dist']": "'.el7'",
}
```

Reply:
If it's just a matter of writing them to file later on, a simple `{opt} = {value}` template will do

Reply:
see above. Could rework this to put the missing ones in a list rather than using one list if you prefer